Route/RoutePageGenerator.php

```diff
@@ -99,9 +99,18 @@ public function update(SiteInterface $site, OutputInterface $output = null)
         }
 
         // Iterate over declared routes from the routing mechanism
+        $routesDedupInfos = array();
+        // We first de-duplicate routes by path; the first declaration win !
         foreach ($this->router->getRouteCollection()->all() as $name => $route) {
-            $name = trim($name);
-
+            $routePath = $route->getPath();
+            if (!array_key_exists($routePath, $routesDedupInfos)) {
```

Comment:
Also, using `current($routesDedupInfos[$routePath])` might be safer and look less arbitrary.